cmd/dagger/cmd/do.go

```diff
@@ -59,7 +59,7 @@ var doCmd = &cobra.Command{
 		daggerPlan, err := loadPlan(viper.GetString("plan"))
 		if err != nil {
 			if viper.GetBool("help") {
-				doHelpCmd(cmd, nil, nil, nil, targetPath, nil)
+				doHelpCmd(cmd, nil, nil, nil, targetPath, []string{err.Error()})
 				os.Exit(0)
```

Comment:
@jlongtine Is it actually alright if we `Exit(0)` even if the plan has problems?

I'm realizing now the logic was on purpose: if `--help` is provided, suppress plan errors, exit 0.

Reply:
Traditionally CLI's don't perform any action if `--help` is provided  even though commands or other flags are specified and they usually return with exit code == 0. They current behavior is what I'd expect naturally when interacting with the CLI.

Having said that, if we want to still validate the plan and return something !=0, I think we should make explicit what the expected behavior is.

Reply:
Hmm... Yeah, that's an interesting case. I think you're correct about why I did it, but I'm not sure what the "right" way to do it would be in this case.

Reply:
I think my personal preference would be to print errors + exit(0) because they're using the `--help` flag.